src/main/java/cz/fi/muni/pa165/Model/Entity/ApplicationApprovedRecord.java

```diff
@@ -59,4 +60,30 @@ public Date getTo() {
     public RentApplication getApplication() {
         return application;
     } 
+    
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if ((o == null) || !(o instanceof ApplicationApprovedRecord)) {
+            return false;
+        }
+        ApplicationApprovedRecord other = (ApplicationApprovedRecord) o;
+        if (getFrom() != null ? !getFrom().equals(other.getFrom()) : other.getFrom() != null) {
+            return false;
+        }
+        if (getTo() != null ? !getTo().equals(other.getTo()) : other.getTo() != null) {
+            return false;
+        }
+        if (application != null ? !application.equals(other.getApplication()) : other.getApplication() != null) {
+            return false;
+        }
+	return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder().append(getFrom()).append(getTo()).append(getApplication().getId()).toHashCode();
```

Comment:
@charlliz link you have sent applies to generic objects. We are talking about entities, which are [defined by identity](http://stackoverflow.com/questions/7941485/what-does-thread-of-continuity-mean-when-describing-an-entity-in-domain-driven-d).

**Consider these facts that applies to our system:**
- we are using UUID -> ID is assigned right after constructor invocation --> there will be never ever entity without ID (this cannot create conflict event in collections)
- Entities are "thread of continuity and identity" thus if two entities has the same ID they are representing the same thing in real world even if there are same changes to some of their values.

**Outcomes of this decision:**
- 👍 Two logical same entities will appear to Java as the same objects (in hash maps and in comparisons) 
- 👍 equals and hash-code generated from ID will work also with detached entities (which we will hopefully not need 😃 )
- 👍  usable in collections (thanks to UUID in constructor)
- 👎 downside: if there will be more [identity maps](https://docs.jboss.org/hibernate/orm/3.5/api/org/hibernate/util/IdentityMap.html): hash-map will need to contain only entities managed by that particular hash-map (hopefully will not need this)
   - BTW hash-map with you current implementation for `hashcode()` will not work either; looks to me like a general limitation

What are you arguments why to keep this in current state?